DEC/src/roe/SSAExtract.jl

```diff
@@ -0,0 +1,137 @@
+module SSAExtract
+
+# 
+using ..DEC: AbstractSort, TypedApplication, TA, Roe, RootVar
+
+# other dependencies
+using MLStyle
+using Metatheory: VecExpr
+using Metatheory.EGraphs
+using StructEquality
+
+"""    SSAVar
+
+A wrapper for the index of a SSAVar
+"""
+@struct_hash_equal struct SSAVar
+    idx::Int
+end
+export SSAVar
+
+function Base.show(io::IO, v::SSAVar)
+    print(io, "%", v.idx)
+end
+
+"""    SSAExpr
+
+A wrapper for a function (::Any) and its args (::Vector{Tuple{Sort, SSAVar}}).
+
+Example: the equation
+```
+  a = 1 + b
+```
+may have an SSA dictionary
+```
+  %1 => a
+  %2 => +(%1, %3)
+  %3 => b
+```
+and so `+` would have
+```
+SSAExpr(+, [(Scalar(), SSAVar(1)), (Scalar(), SSAVar(2))])
+```
+"""
+@struct_hash_equal struct SSAExpr
```

Comment:
Call this `Term` instead of `SSAExpr`, used as `SSAs.Term`. We can't use just Expr because Julia takes that.